### PR TITLE
fix: Fix measuring MPE

### LINKED ISSUE #14490: [Skia.gtk][MediaPlayerElement] The MediaPlayerElement, when given a maximum width, is square, even when space is restricted
### Current behavior

image
The mediaplayer element on skia takes a square shape, instead of following the aspect ratio of the video, when put in an auto row.  (and also, other things that would try to make take the least amount of space as possible)

### Expected behavior

image
On windows and other platform, the mediaplayer keeps the same aspect ratio as the video when put in an auto row. 16/9 in this case

### How to reproduce it (as minimally and precisely as possible)

```xml
<Grid Width="30"
	  Height="720"
	  HorizontalAlignment="Left"
	  VerticalAlignment="Top"
	  Background="BlueViolet" />
<Grid Width="30"
	  Height="405"
	  VerticalAlignment="Top"
	  HorizontalAlignment="Right"
	  Background="Orange" />
<Grid>

	<Grid.RowDefinitions>
		<RowDefinition Height="Auto" />
	</Grid.RowDefinitions>

	<MediaPlayerElement x:Name="MediaPlayerElementSample2"
						Source="ms-appx:///BasicApp/Assets/Videos/Getting_Started_with_Uno_Platform_for_Figma.mp4"
						AreTransportControlsEnabled="True"
						AutoPlay="False"
						MaxWidth="720" />
</Grid>
```

[BasicApp.zip](https://github.com/unoplatform/uno/files/13420799/BasicApp.zip)

### Workaround

_No response_

### Works on UWP/WinUI

Yes

### Environment

Uno.WinUI / Uno.WinUI.WebAssembly / Uno.WinUI.Skia

### NuGet package version(s)

```csproj
<PackageReference Include="Uno.Resizetizer" Version="1.3.0-dev.12" />
<PackageReference Include="Uno.WinUI.Skia.Gtk" Version="5.1.0-dev.542" />
<PackageReference Include="CommunityToolkit.Mvvm" Version="8.1.0" />
<PackageReference Include="Microsoft.Extensions.Logging.Console" Version="7.0.0" />
<PackageReference Include="SkiaSharp.Views.Uno.WinUI" Version="2.88.6" />
<PackageReference Include="SkiaSharp.Skottie" Version="2.88.3" />
<PackageReference Include="Uno.WinUI.RemoteControl" Version="5.1.0-dev.542" Condition="'$(Configuration)'=='Debug'" />
<PackageReference Include="Uno.UI.Adapter.Microsoft.Extensions.Logging" Version="5.1.0-dev.542" />
<PackageReference Include="VideoLAN.LibVLC.Windows" Version="3.0.18" />
<PackageReference Include="Uno.WinUI.MediaPlayer.Skia.Gtk" Version="5.1.0-dev.542" />
```

### Affected platforms

Skia (GTK on Linux/macOS/Windows)

### IDE

Visual Studio 2022

### IDE version

17.7.4

### Relevant plugins

_No response_

### Anything else we need to know?

_No response_

### PR DESCRIPTION
**Pull request by Youssef1313**
GitHub Issue (If applicable): closes #14490

## PR Type

What kind of change does this PR introduce?

### <samp>🤖[[deprecated]](https://githubnext.com/copilot-for-prs-sunset) Generated by Copilot at 52a8293</samp>

This pull request refactors the media player element and its platform-specific implementations to use the natural video dimensions instead of the video ratio to calculate the layout size and to raise events. It also removes or simplifies some unused or redundant code and improves the consistency and naming of the events and properties. The main files affected are `MediaPlayerElement.cs`, `MediaPlayerPresenter.cs`, `MediaPlayer.cs`, and the platform-specific extensions of these classes.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [ ] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: src/Uno.UI/UI/Xaml/Controls/MediaPlayerElement/MediaPlayerPresenter.cs [hunk 1]
```diff
@@ -148,30 +172,90 @@ private void OnSourceChanged(Windows.Media.Playback.MediaPlayer sender, object a
...
+		{
+			if (wrOwner?.TryGetTarget(out var owner) == true && owner is not null && !IsFullWindow)
+			{
+				return owner;
+			}
+
+			return this;
+		}
+
 		protected override Size MeasureOverride(Size availableSize)
```
Comment (Youssef1313) [inline src/Uno.UI/UI/Xaml/Controls/MediaPlayerElement/MediaPlayerPresenter.cs:185]: The logic here is mostly a port from WinUI

File: src/Uno.UI/UI/Xaml/Controls/MediaPlayerElement/MediaPlayerPresenter.cs [hunk 2]
```diff
@@ -9,7 +9,36 @@ namespace Windows.UI.Xaml.Controls
...
+			wrOwner = new WeakReference<MediaPlayerElement>(owner);
+		}
+
+		private float GetScaledOtherDimension(
+			float scaledOneDimension,
+			uint naturalOneDimension,
+			uint naturalOtherDimension)
+		{
+			//
+			// naturalOneDimension is mapped to scaledOneDimension. Map naturalOtherDimension to scaledOtherDimension using the
```
[the remark this answers is not in the document]
Comment (Youssef1313) [reply src/Uno.UI/UI/Xaml/Controls/MediaPlayerElement/MediaPlayerPresenter.cs:25]: Usually when we port code from WinUI, we port the comments as well. The original code is in

**Review by jeromelaban — APPROVED** [review]

**Pull request merged**